backend/test/Test/UploadSpec.hs

```diff
@@ -109,17 +109,15 @@ spec = withContext $ startWithInitial $ do
         onlyDisabled secondaryMeasurements `shouldBe` secondaryDisabled
 
     it "rollbacks targets and compounds due processing error" $ \ctx -> do
+      _ <- ctx <$ runWithInit ctx addInitialData
```

Comment:
`ctx <$ ` causes this expression to return `ctx` and you ignore it. So `ctx <$` is redundant.